users/views_test.py

```diff
@@ -118,3 +121,77 @@ def test_countries_states_view(client):
     assert len(countries.get("FR").get("states")) == 0
     assert countries.get("US").get("name") == "United States"
     assert countries.get("TW").get("name") == "Taiwan"
+
+
+def test_create_email_change_request_invalid_password(user_drf_client, user):
+    """Test that invalid password is returned"""
+    resp = user_drf_client.post(
+        "/api/change-emails/",
+        data={
+            "new_email": "abc@example.com",
+            "password": user.password,
+            "old_password": "abc",
+        },
+    )
+    assert resp.status_code == status.HTTP_400_BAD_REQUEST
+
+
+def test_create_email_change_request_existing_email(user_drf_client, user):
+    """Test that create change email request gives validation error for existing user email"""
+    new_user = UserFactory.create()
+    user_password = user.password
+    user.password = make_password(user.password)
+    user.save()
+    resp = user_drf_client.post(
+        "/api/change-emails/",
+        data={"new_email": new_user.email, "password": user_password},
+    )
+    assert resp.status_code == status.HTTP_400_BAD_REQUEST
+
+
+def test_create_email_change_request_valid_email(user_drf_client, user, mocker):
+    """Test that change request is created"""
+    user_password = user.password
+    user.password = make_password(user.password)
+    user.save()
+
+    mock_response = mocker.patch.object(requests.Session, "get")
+    mock_response.raise_for_status.return_value = None
+    resp = user_drf_client.post(
+        "/api/change-emails/",
+        data={"new_email": "abc@example.com", "password": user_password},
+    )
+
+    assert resp.status_code == status.HTTP_201_CREATED
+
+    code = resp.context[0]["confirmation_url"].split("?verification_code=")[1]
+    assert code
```

Comment:
It looks like this is asserting behavior of `send_verify_email_change_email`? `send_verify_email_change_email` should be mocked instead.

Reply:
Actually, I'm testing the behavior of the email change request here, so by mocking I wouldn't anymore be getting the code so wouldn't be able to test the remaining part of it.